Implementation of rest of rooms, objects, dirObjects to complete Act I

Dkrenon opened the pull request:
-In the **Enums** I added the cellar room, the objects petrol, matchsticks, glue, dynamite, boulder and the action sniff.
-In the **Game Setup System** I implemented the creation of the forge room with its objects, dirObjs, actions, etc. The same for the cellar room. In the mountain pass I added the boulder that will lead to new act.  In the barn I added the stairs with hay that will lead to the cellar and changed the destination of the window from the plain to the forge room. Lastly, I added a variable  when creating the objects that indicates if the object can be picked up instead of assuming that all can be picked up.
-In the **Look System** I changed the Object Description as before it was only functioning for one  object and not multiples. The logic is the same as the DirObject Description yet the only thing changed is that it looks into the objects array instead of the dirObjects array.
-In the ** Tokeniser System** I added the objects and actions needed.
-In the **GameSystems Tests** I added the necessary tests to test the functionality of the new objects as well going into the new rooms.
-In the **nakama/config.yml** file I added a declaration that was added manually to the server that helped with the stuff of refreshing the tokens.

With this the ACT I has been implemented completely

lbdl commented on cardinal/system/system_game_setup.go:
```diff
@@ -124,19 +126,33 @@ func (s *GameSetup) setupBarn(world cardinal.WorldContext) {
 		"window", windowActions, world))
 
+	// KBARN -> DOWN
+	open2Cellar := s.createAction(enums.ActionTypeOpen, " The hay having been burned quickly, reveals the stairs cleared up.", false, false, false, 0, 0)
```
"The hay having burnt fast reveals a set of stairs"

Dkrenon replied:
done

lbdl commented on cardinal/system/system_game_setup.go:
```diff
@@ -124,19 +126,33 @@ func (s *GameSetup) setupBarn(world cardinal.WorldContext) {
+	// KBARN -> DOWN
+	open2Cellar := s.createAction(enums.ActionTypeOpen, " The hay having been burned quickly, reveals the stairs cleared up.", false, false, false, 0, 0)
+
+	burnHay := s.createAction(enums.ActionTypeBurn, " I can hear the crispy sound of the hay burning"+
```
nice:
"You hear the cracking noise of the hay burning quickly"
"as it is consumed by the dark fires"
"sadly this enjoyable moment is short lived"

Dkrenon replied:
done

lbdl commented on cardinal/system/system_game_setup.go:
```diff
@@ -198,6 +227,112 @@ func (s *GameSetup) setupMountainPath(world cardinal.WorldContext) {
+
+	petrolActions := []uint32{burn}
+	objs := []component.Object{s.createObject(enums.ObjectTypePetrol, enums.MaterialTypeIKEA,
+		"a strange liquid that seems to be petrol, probably there is three liters of it, it's highly burnable though", enums.ObjectTypePetrol.String(), petrolActions, true, world)}
```
"... there are about 3 litres of it, its highly flammable"

Dkrenon replied:
done

lbdl commented on cardinal/system/system_game_setup.go:
```diff
@@ -198,6 +227,112 @@ func (s *GameSetup) setupMountainPath(world cardinal.WorldContext) {
+	// Matchsticks Object
+	light := s.createAction(enums.ActionTypeLight, " The matchsticks (despite their small size)"+
+		" lights enough to see a small distance. You have to use them quickly"+
+		" or your fingers will be burn", true, false, true, 0, 0)
```
"or your fingers will get burnt"

Dkrenon replied:
done

lbdl commented on cardinal/system/system_game_setup.go:
```diff
@@ -198,6 +227,112 @@ func (s *GameSetup) setupMountainPath(world cardinal.WorldContext) {
+
+	matchesActions := []uint32{light}
+	objs = append(objs, s.createObject(enums.ObjectTypeMatchsticks, enums.MaterialTypeWood,
+		"a group of matchsticks that have survived the pass of time, you can probably light them up, as they seem to be in a good condition", enums.ObjectTypeMatchsticks.String(), matchesActions, true, world))
```
"a box of matches that ahve survived the passing of the ages..."

Dkrenon replied:
done

lbdl commented on cardinal/system/system_game_setup.go:
```diff
@@ -198,6 +227,112 @@ func (s *GameSetup) setupMountainPath(world cardinal.WorldContext) {
+	// Dynamite Object
+	lightDynamite := s.createAction(enums.ActionTypeLight, " Seeing the fuse sparkling makes you remember the fireworks you loved as a child."+
+		" Suddenly, you return back and see that there is almost no time."+
+		" You have to do something or you will be flying into meat pieces!", true, false, true, 0, 0)
```
" you have to do something or you will be turned to meat puree"

Dkrenon replied:
done

lbdl commented on cardinal/system/system_game_setup.go:
```diff
@@ -198,6 +227,112 @@ func (s *GameSetup) setupMountainPath(world cardinal.WorldContext) {
+		" You have to do something or you will be flying into meat pieces!", true, false, true, 0, 0)
+
+	throwDynamite := s.createAction(enums.ActionTypeThrow, " You throw quickly the dynamite"+
+		" and start running for the hills as your life depens on it", true, false, true, 0, 0)
```
"... depends on it"

Dkrenon replied:
done

lbdl commented on cardinal/system/system_game_setup.go:
```diff
@@ -198,6 +227,112 @@ func (s *GameSetup) setupMountainPath(world cardinal.WorldContext) {
+
+	dynamiteActions := []uint32{lightDynamite, throwDynamite}
+	objs := []component.Object{s.createObject(enums.ObjectTypeDynamite, enums.MaterialTypeIKEA,
+		"a high quality dynamite that can blast almost everything. It needs to be lighted up first", enums.ObjectTypeDynamite.String(), dynamiteActions, true, world)}
```
"high quality (if old) dynamite with a quick fuse. It needs to be lit first"

Dkrenon replied:
done

lbdl commented on cardinal/system/system_game_setup.go:
```diff
@@ -198,6 +227,112 @@ func (s *GameSetup) setupMountainPath(world cardinal.WorldContext) {
+
+	glueActions := []uint32{sniff}
+	objs = append(objs, s.createObject(enums.ObjectTypeGlue, enums.MaterialTypeShit,
+		"some glue of a strange name: &$#${+>@#!@. Don't knowing if it's your imagination, it is calling you", enums.ObjectTypeGlue.String(), glueActions, true, world))
```
"some oddly names glue. Not knowing if its your imagination it seems to be calling you"

Dkrenon replied:
done

lbdl commented on cardinal/t_gameSystems_test.go:
```diff
@@ -1841,6 +1841,412 @@ func TestFetchObjsForType_RoomToInventory_Empty(t *testing.T) {
+	const playerName = "Hueyu"
+	const roomSpawn = 4
+	var tokens1 = []string{"light", "matchsticks"}
+	var expectedOut1 string = ("You Light the Matchsticks. The matchsticks (despite their small size) lights enough to see a small distance. You have to use them quickly or your fingers will be burn.")
```
"you light the matches...."

Dkrenon replied:
done

lbdl commented on cardinal/t_gameSystems_test.go:
```diff
@@ -1841,6 +1841,412 @@ func TestFetchObjsForType_RoomToInventory_Empty(t *testing.T) {
+	var tokens1 = []string{"light", "matchsticks"}
+	var expectedOut1 string = ("You Light the Matchsticks. The matchsticks (despite their small size) lights enough to see a small distance. You have to use them quickly or your fingers will be burn.")
+	var tokens2 = []string{"burn", "petrol"}
+	var expectedOut2 string = ("You Burn the Petrol. The petrol (with its unique funny smell) burns fiercely with a scent that makes you feel dizzy.")
```
"You light the petrol...."

Dkrenon replied:
done

lbdl reviewed (changes requested):
good stuff, just some grammer typos

Dkrenon commented:
With the new push I fixed the grammar stuff, also changed the word matchsticks for matches, and in the enums I changed the strings to be lowercase (football instead of Football, burn instead of Burn) as this was the cause of (You Burn the Petrol). I also corrected the player entity ID in the tests as all the rooms are created before the player is created. The rest of changes are of grammar type

lbdl commented on cardinal/system/system_game_setup.go:
```diff
@@ -244,16 +244,16 @@ func (s *GameSetup) setupForge(world cardinal.WorldContext) {
-	objs = append(objs, s.createObject(enums.ObjectTypeMatchsticks, enums.MaterialTypeWood,
-		"a group of matchsticks that have survived the pass of time, you can probably light them up, as they seem to be in a good condition", enums.ObjectTypeMatchsticks.String(), matchesActions, true, world))
+	objs = append(objs, s.createObject(enums.ObjectTypeMatches, enums.MaterialTypeWood,
+		"a box of matches that ahve survived the passing of the ages, you can probably light them up, as they seem to be in a good condition", enums.ObjectTypeMatches.String(), matchesActions, true, world))
```
"... that have ..."

Dkrenon replied:
done

lbdl commented on cardinal/system/system_game_setup.go:
```diff
@@ -293,22 +293,22 @@ func (s *GameSetup) setupCellar(world cardinal.WorldContext) {
 	glueActions := []uint32{sniff}
 	objs = append(objs, s.createObject(enums.ObjectTypeGlue, enums.MaterialTypeShit,
-		"some glue of a strange name: &$#${+>@#!@. Don't knowing if it's your imagination, it is calling you", enums.ObjectTypeGlue.String(), glueActions, true, world))
+		"some oddly names glue. Not knowing if its your imagination it seems to be calling you", enums.ObjectTypeGlue.String(), glueActions, true, world))
```
"... oddly named ... "

Dkrenon replied:
done

lbdl reviewed (changes requested):
2 small typos. Good stuff!
